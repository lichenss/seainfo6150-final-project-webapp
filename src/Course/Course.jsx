import React from 'react'
import Footer from '../Footer/Footer'
import Header from '../Header/Header'
import styles from './Course.module.css'
import { Link } from 'react-router-dom';

const Course = (props) => {
    var data=props.location.state;
    var {title,imgurl,description}=data;
    return (
        <>
        <Header />
        <section className={styles.section}>
            <div className={styles.container}>
                <div className={styles.imageContainer}>
                    <img
                    className={styles.image}
                    src={data.imgurl}
                    alt={data.title}
                    />
                    </div>
        <div className={styles.titleContainer}>
            <h3 className={styles.title}>{data.title}</h3>
        </div>
        <div className={styles.descriptionContainer}>
            <p>{data.description}</p>
        </div>
        </div>
        <div className={styles.linkContainer}>
        <Link className={styles.link} to={'/registration-confirmation'}>Registration</Link>
        </div>
        </section>
        <Footer />
        </>
    )
}
export default Course
