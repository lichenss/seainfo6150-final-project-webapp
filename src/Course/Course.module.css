.section {
  margin-top: 140px;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
}
.container {
}
.imageContainer {
  width: 100%;
  height: 20rem;
}

.image {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.titleContainer {
}
.title {
}

.descriptionContainer {
}
.linkContainer {
}
.link {
  font-family: "Franklin Gothic Medium", "Arial Narrow", Arial, sans-serif;
  font-size: 3rem;
  color: black;
}

@media screen and (min-width: 500px) {
  .container {
    width: 30rem;
  }
}

@media screen and (min-width: 800px) {
  .container {
    width: 50rem;
  }
}
