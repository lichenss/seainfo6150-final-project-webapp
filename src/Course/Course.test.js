import React from "react";
import Course from "./Course";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
const location={
        state:{
            title:"Therapeutic Art Life Coach Certification (Accredited)",
        imgurl: "https://arhu.umd.edu/sites/default/files/2019-04/iStock-955549248%20composite-JC%20flat2%20opt1-min.jpg",
        description:"Therapeutic Art: Help Clients Heal, Reduce Stress & Manifest Dreams with Mandalas, Affirmations, Vision Boards",
        instructor: "Janet Giessl",
    }
}
describe("Course tests", () => {
    it("renders correctly", () => {
      const { container } = render(
      <BrowserRouter>
      <Course loaction={location} />
      </BrowserRouter>);
      expect(container).toMatchSnapshot();
    });
  });