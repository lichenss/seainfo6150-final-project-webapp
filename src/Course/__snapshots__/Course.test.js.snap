// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Course tests renders correctly 1`] = `
<div>
  <div>
    <header
      class="header"
    >
      <div
        class="navbar"
      >
        <button
          class="logo"
        />
        <div>
          <a
            class="catLink"
            href="/category/computer-science"
          >
            Computer Science
          </a>
          <a
            class="catLink"
            href="/category/business"
          >
            Business
          </a>
          <a
            class="catLink"
            href="/category/data-science"
          >
            Data Science
          </a>
          <a
            class="catLink"
            href="/category/arts-and-humanities"
          >
            Arts and Humanities
          </a>
          <a
            class="catLink"
            href="/category/physical-science-and-engineering"
          >
            Physical Science and Engineering
          </a>
          <a
            class="catLink"
            href="/category/math-and-logic"
          >
            Math and Logic
          </a>
        </div>
        <div
          class="links"
        >
          <a
            class="barLink"
            href="/log-in"
          >
            Log In
          </a>
          <a
            class="barLink"
            href="/sign-up"
          >
            Sign Up
          </a>
        </div>
      </div>
    </header>
  </div>
  <section
    class="section"
  >
    <div
      class="container"
    >
      <div
        class="imageContainer"
      >
        <img
          alt="Therapeutic Art Life Coach Certification (Accredited)"
          class="image"
          src="https://arhu.umd.edu/sites/default/files/2019-04/iStock-955549248%20composite-JC%20flat2%20opt1-min.jpg"
        />
      </div>
      <div
        class="titleContainer"
      >
        <h1
          class="title"
        >
          Therapeutic Art Life Coach Certification (Accredited)
        </h1>
      </div>
      <div
        class="instrcutorContainer"
      >
        <h3>
          Instructor:
        </h3>
        <p
          class="content"
        >
          Janet Giessl
        </p>
      </div>
      <div
        class="descriptionContainer"
      >
        <h3>
          Description:
        </h3>
        <p
          class="content"
        >
          Therapeutic Art: Help Clients Heal, Reduce Stress & Manifest Dreams with Mandalas, Affirmations, Vision Boards
        </p>
      </div>
    </div>
    <div
      class="linkContainer"
    >
      <a
        class="link"
        href="/registration-confirmation"
      >
        Registration
      </a>
    </div>
  </section>
  <footer
    class="footer"
  >
    <div
      class="links"
    >
      <a
        class="link"
        href="/contact-us"
      >
        <h3>
          Contact Us
        </h3>
      </a>
      <a
        class="link"
        href="/about-us"
      >
        <h3>
          About Us
        </h3>
      </a>
    </div>
  </footer>
</div>
`;
